### PR TITLE
#225: DataLayer tests are using the same database concurrently

### LINKED ISSUE #225: DataLayer tests are using the same database concurrently
Some of the DataLayer tests are using the same database concurrently resulting in this warning in our testing logs:
Screenshot 2024-03-26 at 16 21 53
We should be able to switch to generating a unique database (using `uuid`) for each test like done with the Repository tests.

### PR DESCRIPTION
**Pull request by Subzeero**
Closes #225.

# What was the issue
Some of the DataLayer tests were creating both a DataLayer and Repository to conduct testing a database. However, the DataLayer has its own internal Repository, and both of these Repositories were trying to access the database at the same time.

# What was done and why
Spent *way too long* diagnosing the problem and created a method to expose the internal DataLayer Repository instance through the PrivilegedDataLayer for testing. Also setup each test to receive its own database to work in (no more sharing).

# How it was tested
Fixed those tests that weren't correct, and now all tests pass.

### CODE REVIEW

File: my-webxr-app/src/data/DataLayer.tsx [hunk 1]
```diff
@@ -17,9 +17,9 @@ import DataPoint from '../repository/DataPoint';
  * The Data Layer provides a set of methods for working with CSV and PCA data.
  */
 export default class DataLayer implements DataAbstractor {
-  private repository: Repository;
+  protected repository: Repository;
 
-  private isFirstBatch: boolean;
+  protected isFirstBatch: boolean;
```
Comment (trangnguyen3010) [inline my-webxr-app/src/data/DataLayer.tsx:22]: minor: why do we have to change access modifier for isFirstBatch?

**Review by trangnguyen3010 — APPROVED** [review]
💯

**Review by LongQuanNguyen — APPROVED** [review]
Look so good without all the warnings. Loves this, thank for the fix

File: my-webxr-app/src/data/DataLayer.tsx (hunk 1 above)
Comment (Subzeero) [reply my-webxr-app/src/data/DataLayer.tsx:22]: I changed it to be consistent with the above repository variable. I'm not sure we'd ever need to expose this variable, but 🤷‍♂️

Comment (LongQuanNguyen) [reply my-webxr-app/src/data/DataLayer.tsx:22]: we might if we want to reset the flag every time user load another file/ URL. But that is out of scope for now

**Pull request merged**